appinventor/appengine/src/com/google/appinventor/client/youngandroid/TextValidators.java

```diff
@@ -187,15 +188,20 @@ public static String getErrorMessage(String filename){
     String errorMessage = "";
     String noWhitespace = "[\\S]+";
     String firstCharacterLetter = "[A-Za-z].*";
-    if(!filename.matches("[A-Za-z][A-Za-z0-9_]*") && filename.length() > 0) {
-      if(!filename.matches(noWhitespace)) { //Check to make sure that this project does not contain any whitespace
-        errorMessage = MESSAGES.whitespaceProjectNameError();
-      } else if (!filename.matches(firstCharacterLetter)) { //Check to make sure that the first character is a letter
-        errorMessage = MESSAGES.firstCharProjectNameError();
+    String temp = filename.replaceAll("( )+", " ").replace(" ","_");
+    if(temp.length() > 0){
+    if(!temp.matches("[A-Za-z][A-Za-z0-9_]*")) {
+       if (!temp.matches(firstCharacterLetter)) { //Check to make sure that the first character is a letter
+        return MESSAGES.firstCharProjectNameError();
       } else { //The text contains a character that is not a letter, number, or underscore
-        errorMessage = MESSAGES.invalidCharProjectNameError();
+        return MESSAGES.invalidCharProjectNameError();
       }
     }
-    return errorMessage;
+    if (!filename.matches("[A-Za-z][A-Za-z0-9_]*")) { // check to make sure if original filename has no spaces
```

Comment:
See my notes in the other files about considerations of errors vs warnings. Consider moving this check to a getWarningMessage function.